### ISSUE TITLE
Bazarr and Jackett redirect port with subdomain

### ISSUE THREAD

**Issue by altendorfme** [linuxserver/reverse-proxy-confs#223]
Hi!

I use port _8443:443_ in the swag, all applications work correctly except bazarr and jackett as a subdomain.
When I access _bazarr.xyz.com:8443_, it redirects to _bazarr.xyz.com/login_ (without the port).

To fix I changed in the _proxy.conf_ to:
```
proxy_set_header Host $http_host;
proxy_set_header X-Forwarded-Host $http_host;
```

It worked for bazarr, radarr, sonarr, jackett, deluge and portainer

**Comment by nemchik** [comment]
Can you try removing both of the lines you have and use just this one instead?
```nginx
        proxy_set_header X-Forwarded-Host $host:$server_port;
```
Put the line in `proxy.conf` and it should affect all of the proxies that include it.

**Comment by nemchik** [comment]
P.S. we actually already have `proxy_set_header X-Forwarded-Host $host;` in `proxy.conf` but we don't have `:$server_port` on the end. If you can confirm the above line works for you I can PR that change into the swag repo.

**Comment by altendorfme** [comment]
Hi, on bazarr does not work, it redirects to: _https://bazarr:6767/login/_

**Comment by nemchik** [comment]
Can you post you full nginx.conf, proxy.conf, ssl.conf, default site config, and the proxy file your using for bazarr?

**Comment by altendorfme** [comment]
proxy.conf (with my fix)
```
## Version 2020/10/04 - Changelog: https://github.com/linuxserver/docker-swag/commits/master/root/defaults/proxy.conf

# Timeout if the real server is dead
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

# Proxy Connection Settings
proxy_buffers 32 4k;
proxy_connect_timeout 240;
proxy_headers_hash_bucket_size 128;
proxy_headers_hash_max_size 1024;
proxy_http_version 1.1;
proxy_read_timeout 240;
proxy_redirect  http://  $scheme://;
proxy_send_timeout 240;

# Proxy Cache and Cookie Settings
proxy_cache_bypass $cookie_session;
#proxy_cookie_path / "/; Secure"; # enable at your own risk, may break certain apps
proxy_no_cache $cookie_session;

# Proxy Header Settings
proxy_set_header Connection $connection_upgrade;
proxy_set_header Early-Data $ssl_early_data;
proxy_set_header Host $http_host;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header X-Forwarded-Proto https;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Real-IP $remote_addr;
```

proxy.conf (you fix)
```
## Version 2020/10/04 - Changelog: https://github.com/linuxserver/docker-swag/commits/master/root/defaults/proxy.conf

# Timeout if the real server is dead
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

# Proxy Connection Settings
proxy_buffers 32 4k;
proxy_connect_timeout 240;
proxy_headers_hash_bucket_size 128;
proxy_headers_hash_max_size 1024;
proxy_http_version 1.1;
proxy_read_timeout 240;
proxy_redirect  http://  $scheme://;
proxy_send_timeout 240;

# Proxy Cache and Cookie Settings
proxy_cache_bypass $cookie_session;
#proxy_cookie_path / "/; Secure"; # enable at your own risk, may break certain apps
proxy_no_cache $cookie_session;

# Proxy Header Settings
proxy_set_header Connection $connection_upgrade;
proxy_set_header Early-Data $ssl_early_data;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Host $host:$server_port;
proxy_set_header X-Forwarded-Proto https;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Real-IP $remote_addr;
```

bazarr.subdomain.conf
```
# make sure that your dns has a cname set for bazarr and that your bazarr container is not using a base url

server {
    listen 443 ssl;
    listen [::]:443 ssl;

    server_name bazarr.*;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    # enable for ldap auth, fill in ldap details in ldap.conf
    #include /config/nginx/ldap.conf;

    # enable for Authelia
    #include /config/nginx/authelia-server.conf;

    location / {
        # enable the next two lines for http auth
        #auth_basic "Restricted";
        #auth_basic_user_file /config/nginx/.htpasswd;

        # enable the next two lines for ldap auth
        #auth_request /auth;
        #error_page 401 =200 /ldaplogin;

        # enable for Authelia
        #include /config/nginx/authelia-location.conf;
        include /config/nginx/proxy.conf;
        resolver 127.0.0.11 valid=30s;
        set $upstream_app bazarr;
        set $upstream_port 6767;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
    }

    location ~ (/bazarr)?/api {
        include /config/nginx/proxy.conf;
        resolver 127.0.0.11 valid=30s;
        set $upstream_app bazarr;
        set $upstream_port 6767;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;

    }
}
```

ssl.conf
```
## Version 2020/10/04 - Changelog: https://github.com/linuxserver/docker-swag/commits/master/root/defaults/ssl.conf

### Mozilla Recommendations
# generated 2020-06-17, Mozilla Guideline v5.4, nginx 1.18.0-r0, OpenSSL 1.1.1g-r0, intermediate configuration
# https://ssl-config.mozilla.org/#server=nginx&version=1.18.0-r0&config=intermediate&openssl=1.1.1g-r0&guideline=5.4

ssl_session_timeout 1d;
ssl_session_cache shared:MozSSL:10m;  # about 40000 sessions
ssl_session_tickets off;

# intermediate configuration
ssl_protocols TLSv1.2 TLSv1.3;
ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
ssl_prefer_server_ciphers off;

# OCSP stapling
ssl_stapling on;
ssl_stapling_verify on;


### Linuxserver.io Defaults

# Certificates
ssl_certificate /config/keys/letsencrypt/fullchain.pem;
ssl_certificate_key /config/keys/letsencrypt/privkey.pem;
# verify chain of trust of OCSP response using Root CA and Intermediate certs
ssl_trusted_certificate /config/keys/letsencrypt/fullchain.pem;

# Diffie-Hellman Parameters
ssl_dhparam /config/nginx/dhparams.pem;

# Resolver
resolver 127.0.0.11 valid=30s; # Docker DNS Server

# Enable TLS 1.3 early data
ssl_early_data on;

# HSTS, remove # from the line below to enable HSTS
#add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload" always;

# Optional additional headers
#add_header Cache-Control "no-transform" always;
#add_header Content-Security-Policy "upgrade-insecure-requests";
#add_header Referrer-Policy "same-origin" always;
#add_header X-Content-Type-Options "nosniff" always;
#add_header X-Frame-Options "SAMEORIGIN" always;
#add_header X-UA-Compatible "IE=Edge" always;
#add_header X-XSS-Protection "1; mode=block" always;
```

**Comment by nemchik** [comment]
I'm sorry, I misspoke and was unclear.

Please use these lines and see if they work
```nginx
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Host $host:$server_port;
```

The host line is the original default. The forwarded host line is also the original default, but with the port added.

**Comment by altendorfme** [comment]
Okay!

But it didn't work, he keeps removing the port number: _bazarr.xyz.com:8443_, it redirects to _bazarr.xyz.com/login_ (without the port).

proxy.conf
```
## Version 2020/10/04 - Changelog: https://github.com/linuxserver/docker-swag/commits/master/root/defaults/proxy.conf

# Timeout if the real server is dead
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

# Proxy Connection Settings
proxy_buffers 32 4k;
proxy_connect_timeout 240;
proxy_headers_hash_bucket_size 128;
proxy_headers_hash_max_size 1024;
proxy_http_version 1.1;
proxy_read_timeout 240;
proxy_redirect  http://  $scheme://;
proxy_send_timeout 240;

# Proxy Cache and Cookie Settings
proxy_cache_bypass $cookie_session;
#proxy_cookie_path / "/; Secure"; # enable at your own risk, may break certain apps
proxy_no_cache $cookie_session;

# Proxy Header Settings
proxy_set_header Connection $connection_upgrade;
proxy_set_header Early-Data $ssl_early_data;
#proxy_set_header Host $http_host;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
#proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Host $host:$server_port;
proxy_set_header X-Forwarded-Proto https;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Real-IP $remote_addr;
```

**Comment by altendorfme** [comment]
I reinstalled my server and Jackett was not redirecting correctly when using an SSL port (8443 for example).

The solution was to change the `proxy.conf`

From:  `proxy_set_header X-Forwarded-Host $host;`
To: `proxy_set_header X-Forwarded-Host $http_host;`

Or just comment the line `proxy_set_header X-Forwarded-Host $host;`

Bazarr, Plex, Raddar, Sonarr and Jackett worked correctly.

**Comment by nemchik** [comment]
After some additional research I think I see what's happening (although I'm not sure if there's a one-size-fits-all fix).

`$host` will never contain a port number by design.
`$http_host` can contain a port number, and if there is a port number it's pulled from the request header.
`$server_port` is the port defined in the `listen` line(s), ex:
```nginx
    listen 443 ssl;
    listen [::]:443 ssl;
```

So the `$host:$server_port` solution would work as expected if you changed the `listen` port from `443` to `8443` but this would be a change to many files, and also require you set up the container with port 8443 instead of 443.

I don't know if there are any positive or negative effects to using `$http_host` as a permanent solution applied to everyone.

My recommendation for now is for you to ensure you have the latest configs (the container logs will display information about any updates, and the [swag readme](https://github.com/linuxserver/docker-swag/blob/master/README.md) has information about updating the configs) and use `proxy_set_header X-Forwarded-Host $http_host;` as it seems to be working for you. I may consider adding a comment note about this in the `proxy.conf` but leaving the current configs as is.

**Comment by altendorfme** [comment]
Hi, 

Thank you for the informations.

Adding this to the documentation is interesting, it took me a while to find this old comment here myself.
I believe that in many countries ports 80 and 443 are blocked, and this simple modification is useful. ;)

Anyway, in the future it would be interesting to add the HTTP and HTTPs PORTS in the `.env`/`environment`,
I'll take some time to help and create a pull request ;)

**Comment by nemchik** [comment]
Following up on this.

I actually ended up adding the `$server_port` variable to the `X-Forwarded-Host` proxy header here https://github.com/linuxserver/docker-swag/commit/6123952de79654eb3cbbddfb82251b1478346bb4#diff-19a1ac4392b0648872fb48553672f4bb58b932819d5d474e5a63553beb300419R30 in an attempt to allow running SWAG on alternate ports. Unfortunately it was recently brought to my attention that this causes CSRF issues in some cases, such as when using authentication configs also included in SWAG. I removed the `$server_port` variable here https://github.com/linuxserver/docker-swag/commit/3bab8b6b777cba05c11f7765bb9ed0c4841b8171 and added a new header `X-Forwarded-Port` but this may not help with this specific use case (running SWAG on an alternate port).

At this point, our focus is to support running SWAG on the default ports. It should also be possible to use cloudflared https://github.com/linuxserver/docker-mods/tree/universal-cloudflared to circumvent the need to use an alternate port.
